### PR TITLE
optimal filter digitization and fixed some bug

### PR DESCRIPTION
**Pull request by BeigeLiu**
1. add optimal filter module to calculate the optimal filter magnification of signal-noise-ratio
2. add sampling process of the output signal to simulate digitization
3. fixed the bug of _conjugate function in circuit calculator module
4. change the way of calculate ac resistance using recently measured parameters
5. add some files for these modules
6.mind that because github can not pull up 100MB files so the most recent circuit file is not pulled

### CODE REVIEW

**Comment by BeigeLiu** [comment]
magnetometer module, optimal filter module and digitization module are added

File: modules/CircuitCalculator.py [hunk 1]
```diff
@@ -130,7 +122,7 @@ def _generateNoiseFrequencySpec(self, freq, **kwargs):
         '''
         # load info
         Rdc                 = kwargs.get('Rdc', 1) # in Om
-        Rslope              = kwargs.get('Rslope', 1.) # in 1/MHz
+        Rslope              = Rslope_dict(np.abs(freq))
```
Comment (mcfatelin) [inline modules/CircuitCalculator.py:125]: where is the parameters for Rac now?

Comment (mcfatelin) [reply modules/CircuitCalculator.py:125]: why you need a dedicated dictionary to input? why not use existing **kwargs?

**Review by mcfatelin — CHANGES_REQUESTED** [review]

File: modules/CircuitCalculator.py (hunk 1 above)
Comment (BeigeLiu) [reply modules/CircuitCalculator.py:125]: > where is the parameters for Rac now?
there are two noises,and their Rac are different.
in Changqing's file,  Rac is determined by two parameter a and b, but for different noises,a and b has different meaning(different function) , i use interpolation to fit that function.

Comment (BeigeLiu) [reply modules/CircuitCalculator.py:125]: > why you need a dedicated dictionary to input? why not use existing **kwargs?

there will be a bug if using **kwargs cause i do not know how to fix it(or simply i do not really understand how to use **kwargs), so i just add a parameter as input

**Pull request merged**